src/common/preferences/preferences.jsx

```diff
@@ -0,0 +1,618 @@
+		});
+	},
+
+	/**
+	 * Submits an error report
+	 */
+	submitErrors: function() {
+		var reportErrorsButton = document.getElementById('general-button-report-errors');
+		toggleDisabled(reportErrorsButton, true);
+		
+		Zotero.Errors.sendErrorReport(function(status, message) {
+			if(status) {
+				alert('Your error report has been submitted.\n\nReport ID:'+message+'\n\n'+
+					'Please post a message to the Zotero forums (forums.zotero.org) with this Report '+
+					'ID, a description of the problem, and any steps necessary to reproduce it.\n\n'+
+					'Error reports are not reviewed unless referred to in the forums.');
+			} else {
+				alert('An error occurred submitting your error report.\n\n'+message+'\n\n'+
+					'Please ensure that you are connected to the Internet. If the problem persists, '+
+					'please post a message to the Zotero forums (forums.zotero.org).');
+			}
+			toggleDisabled(reportErrorsButton, false);
+		});
+	}
+};
+
+Zotero_Preferences.Components = {};
+
+Zotero_Preferences.Components.ProxyPreferences = React.createClass({
+	getInitialState: function() {
+		let state = {};
+		let settings = ['transparent', 'autoRecognize', 'showRedirectNotification',
+			 'disableByDomain', 'disableByDomainString'];
+		for (let setting of settings) {
+			state[setting] = Zotero.Prefs.get('proxies.'+setting);
+		}
+		
+		return state;
+	},
+	
+	handleCheckboxChange: function(event) {
+		this.updateState(event.target.name, event.target.checked);
+	},
+	
+	handleTextInputChange: function(event) {
+		this.updateState(event.target.name, event.target.value);
+	},
+	
+	updateState: function(name, value) {
+		let newState = {};
+		newState[name] = value;
+		this.setState(newState);
+		Zotero.Prefs.set('proxies.'+name, value);
+		// Refresh prefs in the background page
+		Zotero.Proxies.loadPrefs();
+		if (name === 'transparent') {
+			this.props.onTransparentChange(value);
+		}
+	},
+	
+	render: function() {
+		return (
+			<div>
+				<label><input type="checkbox" name="transparent" onChange={this.handleCheckboxChange} defaultChecked={this.state.transparent}/>&nbsp;Enable proxy redirection</label><br/>
+				<div style={{marginLeft: "1em"}}>
+					<label><input type="checkbox" disabled={!this.state.transparent} onChange={this.handleCheckboxChange} name="autoRecognize" defaultChecked={this.state.autoRecognize}/>&nbsp;Automatically detect new proxies</label><br/>
+					<label><input type="checkbox" disabled={!this.state.transparent} onChange={this.handleCheckboxChange} name="showRedirectNotification" defaultChecked={this.state.showRedirectNotification}/>&nbsp;Show a notification when redirecting through a proxy</label><br/>
+					<br/>
+					<label><input type="checkbox" disabled={!this.state.transparent} onChange={this.handleCheckboxChange} name="disableByDomain" defaultChecked={this.state.disableByDomain}/>&nbsp;Disable proxy redirection when my domain name contains (only available when Zotero Client is running)</label><br/>
```

Comment:
> Disable proxy redirection when my domain name contains (only available when Zotero Client is running)

I think the part in parens has to be an asterisk below or somehow otherwise separate from this line

Also just "Zotero", not "Zotero Client"